src/InteractiveWindow/Editor/Commands/InteractiveWindowCommands.cs

```diff
@@ -237,40 +239,44 @@ public Task<ExecutionResult> TryExecuteCommand()
-"                     " + InteractiveWindowResources.UpArrowHelp2,
-"DownArrow            " + InteractiveWindowResources.DownArrowHelp1,
-"                     " + InteractiveWindowResources.DownArrowHelp2,
-"Ctrl-K, Ctrl-Enter   " + InteractiveWindowResources.CtrlKCtrlEnterHelp,
-"Ctrl-E, Ctrl-Enter   " + InteractiveWindowResources.CtrlECtrlEnterHelp,
-"Ctrl-A               " + InteractiveWindowResources.CtrlAHelp,
-        };
-
         private static readonly string[] s_CSVBScriptDirectives = new[]
         {
 "#r                   " + InteractiveWindowResources.RefHelp,
 "#load                " + InteractiveWindowResources.LoadHelp
         };
 
+        private static readonly string s_shortcutDescriptionsFormat =
+            HelpIndent + "Enter                " + InteractiveWindowResources.EnterHelp            + "\r\n" +
+            HelpIndent + "Ctrl-Enter           " + InteractiveWindowResources.CtrlEnterHelp1       + "\r\n" +
+            HelpIndent + "                     " + InteractiveWindowResources.CtrlEnterHelp2       + "\r\n" +
+            HelpIndent + "Shift-Enter          " + InteractiveWindowResources.ShiftEnterHelp       + "\r\n" +
+            HelpIndent + "Escape               " + InteractiveWindowResources.EscapeHelp           + "\r\n" +
+            HelpIndent + "Alt-UpArrow          " + InteractiveWindowResources.AltUpArrowHelp       + "\r\n" +
+            HelpIndent + "Alt-DownArrow        " + InteractiveWindowResources.AltDownArrowHelp     + "\r\n" +
+            HelpIndent + "Ctrl-Alt-UpArrow     " + InteractiveWindowResources.CtrlAltUpArrowHelp   + "\r\n" +
+            HelpIndent + "Ctrl-Alt-DownArrow   " + InteractiveWindowResources.CtrlAltDownArrowHelp + "\r\n" +
+            "{0}" +
+            HelpIndent + "Ctrl-K, Ctrl-Enter   " + InteractiveWindowResources.CtrlKCtrlEnterHelp   + "\r\n" +
+            HelpIndent + "Ctrl-E, Ctrl-Enter   " + InteractiveWindowResources.CtrlECtrlEnterHelp   + "\r\n" +
+            HelpIndent + "Ctrl-A               " + InteractiveWindowResources.CtrlAHelp            + "\r\n";
+
+        private static readonly string s_shortcutDescriptionsSmartUpDown = 
+            string.Format(s_shortcutDescriptionsFormat,
+                          HelpIndent + "UpArrow              " + InteractiveWindowResources.UpArrowHelp1   + "\r\n" +
+                          HelpIndent + "                     " + InteractiveWindowResources.UpArrowHelp2   + "\r\n" +
+                          HelpIndent + "DownArrow            " + InteractiveWindowResources.DownArrowHelp1 + "\r\n" +
+                          HelpIndent + "                     " + InteractiveWindowResources.DownArrowHelp2 + "\r\n");
+
+        private static readonly string s_shortcutDescriptionsNoSmartUpDown =
+            string.Format(s_shortcutDescriptionsFormat, string.Empty);
+
+
+        internal string ShortcutDescriptions => UseSmartUpDown ? s_shortcutDescriptionsSmartUpDown : s_shortcutDescriptionsNoSmartUpDown; 
```

Comment:
I just realized the option of UseSmartUpDown can be set/reset at anytime by interactive window, so we have to check everytime `DisplayHelp` is executed.